service/controller/v21/resource/cloudformation/update.go

```diff
@@ -27,6 +28,25 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 		return microerror.Mask(err)
 	}
 
+	// Update Status.Cluster.Scaling.DesiredCapacity before proceeding with
+	// CloudFormation stack updates in case it fails for some reason so current
+	// status is still updated periodically and therefore can provide little
+	// bit more insight into cluster state despite of possible temporary
+	// problem.
+	{
+		newObj, err := r.g8sClient.ProviderV1alpha1().AWSConfigs(customObject.GetNamespace()).Get(customObject.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		newObj.Status.Cluster.Scaling.DesiredCapacity = stackStateToUpdate.WorkerDesired
+
+		_, err = r.g8sClient.ProviderV1alpha1().AWSConfigs(newObj.Namespace).UpdateStatus(newObj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Should the reconciliation be cancelled here? New update event is about to arrive.

Reply:
I'm not actually 100% sure about this. When I was writing this my feeling was that no matter what you do here, it would get into infinite loop, but that's not the case. When cancelling reconciliation and working on updated version, wouldn't it end up to this same state? I kind of stopped worrying about this after I saw in practice that this was working correctly as-is (meaning it updates status according to refresh period of 5min).

Reply:
I would make sense to me if `UpdateStatus` wouldn't generate an event, but I thought it isn't the case.

Reply:
This is an ongoing issue I cannot wrap my head around. Like, how should the reconciliation work in a perfect world? You maybe only want to react on changes to the metadata, spec, or status. All other triggers can be ignored in some cases. So the framework should maybe support something like this, which is probably a bit difficult to implement and maybe not even the right place in the framework. I am wondering if upstream is interested in such feature.

Reply:
And at the end we probably want to have separate resources to reflect the separate logic steps. We want to break down the CF resource anyway.

Reply:
> Like, how should the reconciliation work in a perfect world? You maybe only want to react on changes to the metadata, spec, or status.

Let's narrow the scope to events generated by updating `status` part of the object which is the case here. Should a resource always cancel reconciliation after doing so? I personally think the resource shouldn't update status more than once per reconciliation loop and yes it should always cancel reconciliation after updating status. Is there any scenario you see where that would be disruptive?

Reply:
I have nothing in mind concerning disruptions. Agree on cancelation.